README.md

```diff
@@ -1,50 +1,141 @@
+
+Whenever a network is created it also deploys a N number of PoETs. The number of poets can be specified using CLI option.  The PoET are assigned to the nodes in round robin fashion using `poet-server` config of go-spacemesh. The first node is then added to the config file so home smeshers use the first PoET always. 
+
+The bootnodes are always tied to a particular k8s worker node in round robin fashion. There are mainly two reasons for doing this: 
+
+- Firstly if we don't do this then its possible k8s will host multiple bootnodes in one k8s worker node and if that VM crashes then all bootnodes are down and new smeshers cannot join the network.
+- Secondly the TCP port of the bootnodes needs to be bind'ed to the host node otherwise p2p discovery fails. p2p discover doesn't work over k8s NodePort service. Therefore if we don't bind bootnodes to host node then the k8s node in which the bootnodes are hosted can be rescheduled and therefore the IP to reach the bootnodes will change and smeshers will not be able to connect to the all or some bootnodes.
+
+Spacecraft calculates the total number of k8s nodes need to be created dynamically based on total pods and their resource size. 
+
+The GRPC and JSON API ports of all the managed miners are exposed as separate NodePort. You can get list of all the managed miners GRPC URLs using the `hosts` sub-command.
+
+## Logs
+
+Spacecraft deploys ELK stack for aggregation of logs. It uses filebeat to collect logs and directly stores them in Elasticsearch. All the fields in the logs are converted to string and non-JSON logs are stored raw. There is no logstash deployed because logstash is slow at processing logs instead it uses filebeat logs processing scripts which can process logs  in parallel and very fast.
+
+To deploy filebeat, elasticsearh and kibana it uses helm charts by elastic ([https://github.com/elastic/helm-charts](https://github.com/elastic/helm-charts)).
+
+## Monitor and Alerts
+
+Spacecraft deploys spacemesh-watch ([https://github.com/spacemeshos/spacemesh-watch](https://github.com/spacemeshos/spacemesh-watch)) service which uses GRPC API of the managed miners to monitor them. And when some issue is detected it send alerts in slack. 
+
+## Pyroscope
+
+Pyroscope is an continuous profiling platform. Spacecraft also deploys pyroscope for debugging performance related issues of miners. Pyroscope doesn't support horizontal scaling therefore we cannot collect data of all the miners so we only collect data of miner-10 and miner-20.
```

Comment:
Is the `miner-10`, `miner-20` selection static/hardcoded?

Reply:
Yes hardcoded. I have updated the readme.